app/controllers/api/reviews_controller.rb

```diff
@@ -0,0 +1,46 @@
+class Api::ReviewsController < ApiController
+
+  def create
+    user = User.find_by_name(params['user_name'])
+
+    unless user
+        user = User.create({ :name => params['user_name']})
+
+        unless user
+          json_error(error: 'USER_CREATE_FAILED', error_message: 'Unable to save user')
+        end
+    end
+
+    review = Review.new({
+       :user=> user,
+       :restaurant_id => params['restaurant'],
+       :rating => params['rating'],
+       :message => params['message'],
+    })
+
+    if review.save
+      json_response(review)
+    else
+      json_error(error: 'REVIEW_CREATE_FAILED', error_message: 'Unable to create review')
+    end
+  end
+
+  def update
+    review = Review.find(params[:id])
```

Comment:
I recommend using `params.require(:id)`. In that case, if that parameter is missing, you get an exception _before_ the database query is sent.

Reply:
The Id is part of the url schema it must be included